Helper/PCAHelper.cs

```diff
@@ -0,0 +1,220 @@
+    /// before excute is called.
+    /// </summary>
+    public class PCAHelper : IPCAHelper
+    {
+        /// <summary>
+        /// Instance of the helper
+        /// </summary>
+        public static IPCAHelper Instance { get; private set; }
+
+        /// <summary>
+        /// IPublicClientApplication that is created on the first get
+        /// If you want to customize PublicClientApplicationBuilder, please do it before calling the first get
+        /// </summary>
+        public IPublicClientApplication PCA
+        {
+            get
+            {
+                if (_pca == null)
+                {
+                    _pca = PCABuilder.Build();
+                }
+
+                return _pca;
+            }
+        }
+
+        // Instance of IPublicClientApplication
+        private IPublicClientApplication _pca;
+
+        /// <summary>
+        /// Application builder. It is created in Init and this member can be customized before Build occurs in PCA->get
+        /// </summary>
+        public PublicClientApplicationBuilder PCABuilder { get; private set; }
+
+        /// <summary>
+        /// This is applicable to Android. Please update this property in MainActivity.Create method
+        /// and consequently with change in the current activity.
+        /// </summary>
+        public object ParentWindow { get; set; } = null;
+
+        /// <summary>
+        /// In UWP app, set it to true.
+        /// </summary>
+        public bool IsUWP { get; set; } = false;
+
+        /// <summary>
+        /// This stores the authentication result, from the auth process.
+        /// When the process starts, it is set to null.
+        /// </summary>
+        public AuthenticationResult AuthResult { get; internal set; }
+
+        // desired scopes
+        private string[] _scopes;
+
+        /// <summary>
+        /// Initializes the PCAHelpr or its derived class as per the generics and PublicClientApplicationBuilder with the given parameters
+        /// PublicClientApplicationBuilder can be customized after this method prior to accessing PublicClientApplication.
+        /// By default it is singleton pattern with option to force creation. 
+        /// </summary>
+        /// <typeparam name="T">Any class that is inherited from PCAHelper</typeparam>
+        /// <param name="clientId">Client id of your application</param>
+        /// <param name="scopes">The desired scope</param>
+        /// <param name="specialRedirectUri">If you are using recommended pattern fo rredirect Uri, this is optional</param>
+        /// <param name="forceCreate">Creates a new instance irrespective of the existance of the previous instance</param>
+        /// <returns>Instance of class inherited from PCAHelper</returns>
+        public static IPCAHelper Init<T>(string clientId, string[] scopes, string specialRedirectUri = null, bool forceCreate = false) 

```

Comment:
scopes are per token request, so you could do `GetTokenForUser(string[] scopes, string tenantId = null)`...maybe add other options, like force refresh. This is similar to the higher level API we have in Id. Web, and it would be nice to have consistency, if possible.

Reply:
Good catch.